resolwe/flow/managers/listener.py

```diff
@@ -211,6 +211,108 @@ def hydrate_spawned_files(self, exported_files_mapper, filename, data_id):
 
         return {"file_temp": export_fn, "file": filename}
 
+    def handle_annotate(self, obj):
+        """Handle an incoming ``Data`` object annotate request.
+
+        :param obj: The Channels message object. Command object format:
+
+            .. code-block:: none
+
+                {
+                    'command': 'annotate',
+                    'data_id': [id of the :class:`~resolwe.flow.models.Data`
+                               object this command annotates],
+                    'annotations': {
+                        [annotations to be added/updated]
+                    }
+                }
+        """
+        def report_failure():
+            async_to_sync(self._send_reply)(
+                obj, {ExecutorProtocol.RESULT: ExecutorProtocol.RESULT_ERROR}
+            )
+            async_to_sync(consumer.send_event)(
+                {
+                    WorkerProtocol.COMMAND: WorkerProtocol.ABORT,
+                    WorkerProtocol.DATA_ID: obj[ExecutorProtocol.DATA_ID],
+                    WorkerProtocol.FINISH_COMMUNICATE_EXTRA: {
+                        "executor": getattr(settings, "FLOW_EXECUTOR", {}).get(
+                            "NAME", "resolwe.flow.executors.local"
+                        ),
+                    },
+                }
+            )
+        
+        data_id = obj[ExecutorProtocol.DATA_ID]
+        annotations = obj[ExecutorProtocol.ANNOTATIONS]
+
+        logger.debug(
+            __("Handling annotate for Data with id {} (handle_annotate).", data_id),
+            extra={"data_id": data_id, "packet": obj},
+        )
+        try:
+            d = Data.objects.get(pk=data_id)
+        except Data.DoesNotExist:
+            logger.warning(
+                "Data object does not exist (handle_annotate).",
+                extra={"data_id": data_id,},
+            )
+            report_failure()
+            return
+
+        if d.entity is None:
+            logger.error(
+                __(
+                    "No entity to annotate for process '{}' (handle_annotate):\n\n{}",
+                    d.process.slug,
+                    traceback.format_exc(),
+                ),
+                extra={"data_id": data_id},
+            )
+            d.process_error.append("No entity to annotate for process '%s' (handle_annotate)" % d.process.slug)
+            d.status = Data.STATUS_ERROR
+            try:
+                d.save(update_fields=["process_error", "status"])
+            except Exception:
+                pass
+            report_failure()
+            return
+
+        for key, val in annotations.items():
+            logger.debug(
+                __(
```

Comment:
Put` __("Annotating entity {}: {} -> {}", d.entity, key, val)` in a single line. And probably even `logger.debug` would fit into the same line.